src/main/scala/shine/cuda/primitives/functional/Map.scala

```diff
@@ -0,0 +1,58 @@
+package shine.cuda.primitives.functional
+
+import shine.DPIA.Compilation.TranslationContext
+import shine.DPIA.Compilation.TranslationToImperative._
+import shine.DPIA.DSL._
+import shine.DPIA.Phrases._
+import shine.DPIA.Semantics.OperationalSemantics
+import shine.DPIA.Semantics.OperationalSemantics._
+import shine.DPIA.Types.DataType._
+import shine.DPIA.Types._
+import shine.DPIA._
+import shine.OpenCL.ParallelismLevel
+import shine.cuda.primitives.intermediate.MapI
+import shine.macros.Primitive.expPrimitive
+
+@expPrimitive
+final case class Map(level: ParallelismLevel,
```

Comment:
is this different from the OpenCL map and can we merge the two primitives?

Reply:
This is the same as in OpenCL using the cuda.MapI instead of opencl.MapI to generate code for mapBlock, mapThreads, ... instead of mapWorkgroup and mapLocal.
This primitives should be eventually merged.

Reply:
mentioned in #106